### PR TITLE
[WIP] Feature/instances with numbers

### PR DESCRIPTION
**Pull request by gbourguignon**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/instance/Instance.scala [hunk 1]
```diff
@@ -237,13 +237,14 @@ object Instance {
     }
   }
 
-  case class Id(val runSpecId: PathId, val prefix: Prefix, uuid: UUID) extends Ordered[Id] {
+  case class Id(val runSpecId: PathId, val prefix: Prefix, uuid: UUID, instanceNumber: Int = 0) extends Ordered[Id] {
     lazy val safeRunSpecId = runSpecId.safePath
-    lazy val executorIdString: String = prefix + safeRunSpecId + "." + uuid
+    val numberPrefix = if (instanceNumber > 0) "" + instanceNumber + "-" else ""
```
Comment (komuta) [inline src/main/scala/mesosphere/marathon/core/instance/Instance.scala]: Replace "-" by ".". Considering current heuristics used to extract app name / uuid, this is your best chance to be less intrusive.

Comment (gbourguignon) [reply src/main/scala/mesosphere/marathon/core/instance/Instance.scala]: already done

Comment (gbourguignon) [reply src/main/scala/mesosphere/marathon/core/instance/Instance.scala]: current code: 
 val numberPrefix = if (instanceNumber > 0) f"${instanceNumber}%04d." else ""

**Comment by kamaradclimber** [comment]
Let's close this PR since we don't plan to continue to work on this for the moment.

**Pull request closed without merging**